### PR TITLE
[Artwork] Show RN version when non-commercial only.

### PR DESCRIPTION
**Pull request by alloy**
⚠️ @ashfurrow We should make sure not to include this in the 5.0.3 update. I didn’t create a new CHANGELOG version yet, because I’m not sure what the status is yet. Should we create and target a new `react-native-artwork` branch, for now?

Closes https://artsyproduct.atlassian.net/browse/PURCHASE-1103

Note that this does not yet put showing the RN-based VC behind a lab option, we'll do that in a follow-up PR.

cc @ds300 

#trivial

### CODE REVIEW

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 1]
```diff
@@ -1,334 +1,111 @@
 #import "ARArtworkViewController.h"
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: It's hard to read this diff, but essentially we’ve moved all of the old artwork VC code into a `ARLegacyArtworkViewController` and made the `ARArtworkViewController` only be a parent VC that conditionally shows the legacy or the new RN-based artwork VC.

See https://github.com/artsy/eigen/blob/9d5eeb0b9e2cb93dd87d9b42f3e3ad95015f66f9/Artsy/View_Controllers/Artwork/ARArtworkViewController.m

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 2]
```diff
@@ -1,334 +1,111 @@
...
+@property (strong, nonatomic) ARSpinner *spinner;
 @end
 
-
 @implementation ARArtworkViewController
 
-@dynamic view;
-
-- (void)dealloc;
+// TODO What do we do in the case of loading an artwork in the context of a Fair?
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: @ds300 Did you update any of the tickets to make sure we track this question?

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 3]
```diff
@@ -1,334 +1,111 @@
...
-        // * https://github.com/artsy/eigen/pull/218#issuecomment-75958606
-        self.view.delegate = nil;
-        // And nill-ify these as well, for good measure.
-        self.view.metadataView.delegate = nil;
-        self.view.artworkBlurbView.delegate = nil;
-        self.postsVC.delegate = nil;
-    }
+  return self.fair == nil
+  && (self.artwork.availability == ARArtworkAvailabilityNotForSale
+      || self.artwork.availability == ARArtworkAvailabilitySold);
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: See criteria here https://artsyproduct.atlassian.net/browse/PURCHASE-1042

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 4]
```diff
@@ -1,334 +1,111 @@
...
-        [self.view.stackView removeSubview:self.postsVC.view];
-        _postsVC = nil;
-    }
-}
-
-- (NSString *)inquiryURLRepresentation
-{
-    return [NSString stringWithFormat:@"http://artsy.net/artwork/%@", self.artwork.artworkID];
-}
+  [self.artwork onArtworkUpdate:^{
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: First we let Eigen update the artwork, to ensure we have the `availability` data…

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 5]
```diff
@@ -1,334 +1,111 @@
...
-    NSTimeInterval interval = [saleArtwork.auction.uiDateOfInterest timeIntervalSinceDate:now];
-    if (interval > 0) {
-        self.updateInterfaceWhenAuctionChangesTimer = [NSTimer scheduledTimerWithTimeInterval:interval target:self selector:@selector(reloadUI) userInfo:nil repeats:NO];
-    }
-}
+      [sself.spinner stopAnimating];
+      [sself.spinner removeFromSuperview];
 
-#pragma mark - ScrollView delegate methods
+      if (sself.shouldShowNewVersion) {
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: …then, depending on criteria linked from `shouldShowNewVersion`, we…

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 6]
```diff
@@ -1,334 +1,111 @@
...
+        ARArtworkComponentViewController *vc = [[ARArtworkComponentViewController alloc] initWithArtworkID:sself.artwork.artworkID];
+        [sself addChildViewController:vc];
+        [sself.view addSubview:vc.view];
+        [vc.view alignToView:sself.view];
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: …show the RN-based version, which will end up making its own network request (i.e. non of the data fetched before is re-used, as mentioned [here](https://artsyproduct.atlassian.net/browse/PURCHASE-1042?focusedCommentId=20381&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20381).)

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 7]
```diff
@@ -1,334 +1,111 @@
...
+      } else {
+        // Pass a copy of the artwork instance here, because we don’t want our
+        // artwork's onArtworkUpdate deferred callback to be triggered again.
+        sself.legacyViewController = [[ARLegacyArtworkViewController alloc] initWithArtwork:[sself.artwork copy]
+                                                                                       fair:sself.fair];
+        [sself addChildViewController:sself.legacyViewController];
+        [sself.view addSubview:sself.legacyViewController.view];
+        [sself.legacyViewController.view alignToView:sself.view];
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: …or we show the legacy version with a copy of the artwork instance that already has been full loaded (no further data fetching is needed).

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 8]
```diff
@@ -1,334 +1,111 @@
...
+      }
+    });
+  } failure:^(NSError *error) {
+    // TODO
+    NSLog(@"FAILED TO UPDATE: %@", error);
```
Comment (alloy) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: The comment says it all.

**Comment by alloy** [comment]
Tests are failing, which makes total sense and we should update/add test coverage tomorrow morning.

**Comment by ashfurrow** [comment]
I've created the `react-native-artwork` branch and have protected it from being deleted in the repo settings. I'll take a look at this PR now 👍

File: Podfile [hunk 9]
```diff
@@ -80,7 +80,7 @@ target 'Artsy' do
   pod 'Artsy+UILabels'
   pod 'Extraction'
 
-  pod 'Emission', '~> 1.0'
+  pod 'Emission', '~> 1.11.0'
```
Comment (ashfurrow) [inline Podfile]: I'm curious what you think about the extra specificity here? I think we took this out during City Guide development – I'm fine either way.

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m [hunk 10]
```diff
@@ -1,334 +1,111 @@
...
+- (void)setIndex:(NSInteger)index;
 {
-    return [UIDevice isPad];
+  self.legacyViewController.index = index;
 }
 
-#pragma mark rotation
-
-- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
+- (UIImageView *)imageView;
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: Haha, I have no allies left against ending method implementation names with `;`

Comment (alloy) [reply Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: 😂

File: Podfile (hunk 9 above)
Comment (alloy) [reply Podfile]: The patch component should probably not be needed indeed 👍

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m (hunk 2 above)
Comment (ds300) [reply Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: We resolved this on Jira https://artsyproduct.atlassian.net/browse/PURCHASE-1042

tldr; we won't show the new artwork page for artworks in fairs.

File: Artsy/View_Controllers/Artwork/ARArtworkViewController.m (hunk 10 above)
Comment (alloy) [reply Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: They are still needed in headers and–as I don’t know Xcode editor shortcuts–this keeps copy-pasting easy.

Comment (ashfurrow) [reply Artsy/View_Controllers/Artwork/ARArtworkViewController.m]: Fair enough!

**Comment by alloy** [comment]
Merge on green
